modules/caddyhttp/opentelemetry/module.go

```diff
@@ -0,0 +1,123 @@
+package opentelemetry
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+	"github.com/caddyserver/caddy/v2/caddyconfig/httpcaddyfile"
+	"github.com/caddyserver/caddy/v2/modules/caddyhttp"
+	"go.uber.org/zap"
+)
+
+func init() {
+	caddy.RegisterModule(OpenTelemetry{})
+	httpcaddyfile.RegisterHandlerDirective("opentelemetry", parseCaddyfile)
```

Comment:
Naming-wise, I think it may be better to use a more generic term like `tracing`. We're not likely to support any other tracing APIs or SDKs. The OpenTelemetry project also supports more than just tracing, and I think it would be best to explicitly focus this module on tracing:

```suggestion
	caddy.RegisterModule(Tracing{})
	httpcaddyfile.RegisterHandlerDirective("tracing", parseCaddyfile)
```